code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -386,8 +387,46 @@
 				M.contract_disease(D)
 			else //Injected
 				M.contract_disease(D, 1, 0)
+	//----------------------------------------------------
 
-	if(iscarbon(M) || ismouse(M))
+	if(iscarbon(M))
+		var/mob/living/L = M
+		if(self.data && self.data["virus2"]) //Infecting
+			var/list/blood_viruses = self.data["virus2"]
+			if (istype(blood_viruses) && blood_viruses.len > 0)
+				for (var/ID in blood_viruses)
+					var/datum/disease2/disease/D = blood_viruses[ID]
+					if(method == TOUCH)
+						var/block = L.check_contact_sterility(FULL_TORSO)
+						var/bleeding = L.check_bodypart_bleeding(FULL_TORSO)
+						if (!block)
+							if (D.spread & SPREAD_CONTACT)
+								L.infect_disease2(D, notes="(Contact, splashed with infected blood)")
+							else if (bleeding && (D.spread & SPREAD_BLOOD))
+								L.infect_disease2(D, notes="(Blood, splashed with infected blood)")
+					else
+						L.infect_disease2(D, 1, notes="(Drank/Injected with infected blood)")
+
+		if(ishuman(L) && (method == TOUCH))
+			var/mob/living/carbon/human/H = L
+			H.bloody_body(self.data["donor"])
+			if(self.data["donor"])
+				H.bloody_hands(self.data["donor"])
+			spawn() //Bloody feet, result of the blood that fell on the floor
```

Comment:
I'm not sure if I fully understand this block.
Is the goal to always ensure the human has bloody feet ?

Reply:
This block of code is useless since reaction_animal() cannot have humans as parameter, I forgot to remove it when copying the code from reaction_mob (which by the way doesn't affect simple animals at all as I found out).

Anyway in reaction_mob() the point of this block is to ensure that if you're getting splashed with blood (not injected/drinking) you'll have bloody feets and leave a trail of footprints.